tools/functional-tester/etcd-tester/key_stresser.go

```diff
@@ -202,6 +210,87 @@ func newStressPut(kvc pb.KVClient, keySuffixRange, keySize int) stressFunc {
 	}
 }
 
+func newStressTxn(kvc pb.KVClient, keyTxnSuffixRange, depth int) stressFunc {
+	keys := make([]string, keyTxnSuffixRange)
+	for i := range keys {
+		keys[i] = fmt.Sprintf("/k%03d", i)
+	}
+	for i, k := range keys {
+		_, err := kvc.Put(context.Background(), &pb.PutRequest{
+			Key:   []byte(k),
+			Value: []byte{byte(i)},
+		}, grpc.FailFast(false))
+		if err != nil {
+			plog.Fatalf("newStressTxn error (%v)", err)
+		}
+	}
+	return writeTxn(kvc, keys, depth)
+}
+
+func writeTxn(kvc pb.KVClient, keys []string, depth int) stressFunc {
+	return func(ctx context.Context) (error, int64) {
+		ks := make(map[string]struct{}, depth)
+		for len(ks) != depth {
+			ks[keys[rand.Intn(64)]] = struct{}{}
+		}
+		selected := make([]string, 0, depth)
+		for k := range ks {
+			selected = append(selected, k)
+		}
+		com, delOp, putOp := getTxnReqs(selected[0], "bar00")
+		txnReq := &pb.TxnRequest{
+			Compare: []*pb.Compare{com},
+			Success: []*pb.RequestOp{delOp},
+			Failure: []*pb.RequestOp{putOp},
+		}
+
+		// add nested txns if any
+		for i := 1; i < depth; i++ {
```

Comment:
I dont think i understand the depth here.

it seems to me that `len(txn.Success)` or `len(txn.Failure)` == depth. e.g
`txnReq.Success[nested, nested, nested]`

I thought depth means
`txnReq.Success[nested[nested[nested]]]`

maybe i am missing something.